File: 백준/Bronze/1978. 소수 찾기/소수 찾기.js
const input = require('fs')
  .readFileSync(process.platform === 'linux' ? '/dev/stdin' : './input.txt')
  .toString()
  .trim()
  .split('\n');

const N = Number(input[0]);
const number = input[1].split(' ').map(Number);

let sum = 0;

for (let i = 0; i < N; i++) {
  if (number[i] < 2) continue;

  let isPrime = true;

  for (let j = 2; j * j <= number[i]; j++) {
    if (number[i] % j === 0) {
      isPrime = false;
    }
  }

  if (isPrime) ++sum;
}

console.log(sum);

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import sys

N = int(sys.stdin.readline())
numbers = list(map(int, sys.stdin.readline().split()))

MAX = max(numbers)
isPrime = [True] * (MAX +1)
isPrime[0] = isPrime[1] = False

for i in range (2, int(MAX**0.5)+1):
    if isPrime[i]:
        for j in range(i*i, MAX+1, i):
            isPrime[j] = False
            
count = 0
for n in numbers:
    
    if isPrime[n]:
        count += 1

print(count)